### DISCUSSION TITLE
useResponseCache and RateLimiting

### DISCUSSION THREAD

**Discussion by MDrooker** [dotansimha/envelop#1757] [Q&A] [answered]
Anyone have advice in using ```useResponseCache``` but also needing to enforce rateLimiting at also?

In my testing, ```useRsponseCache``` doesnt let the request deeper in the parse tree to allow rateLimiting to get called.

So seems like we can have one or the other..but not both.

Am I missing something?

**Comment by MDrooker** [comment]
NM-
The default behavior of the ```shouldCacheResult``` was set to true... So thats why the above was happening.
But-
It would be great to also have the context passed into the ```shouldCacheResult``` for some introspection .
I see its passed into the ```enabled``` function.  Is there a way to pass it into the ```shouldCacheResult``` function also?

**Reply by n1ru4l** [reply, marked as answer]
Hey @MDrooker, sorry for the late response. Right now we do not pass in the context there. However, it could make quite sense to also give access there. Would you be willing to open a PR for doing that?
